bridge-sdk/bridge-clients/solana-bridge-client/src/solana_bridge_client.rs

```diff
@@ -220,6 +221,36 @@ impl SolanaBridgeClient {
             .await
     }
 
+    pub async fn deploy_token_with_event(
```

Comment:
Should we move event parsing outside of solana client and only pass the payload? Perhaps even the extraction of the payload should be done in the near client

Reply:
I don't think that I understand what do you mean. There's no event parsing in solana client, I provide this method directly with Nep141Event like we do with evm methods

Reply:
I mean the part where we extract payload and signature from the event. Seems unnecessary for a Solana module to know about the structure of Near events

Reply:
Oh, yes, that makes sense. Thanks for pointing that out
Fixed in https://github.com/Near-One/bridge-sdk-rs/pull/59/commits/b3b854050d40cac6ac06e35e528448f6652d3f2f